utxo/src/undo.rs

```diff
@@ -14,10 +14,22 @@
 // limitations under the License.
 
 use crate::Utxo;
-use common::primitives::BlockHeight;
 use serialization::{Decode, Encode};
 
-#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+#[derive(Default, Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct BlockRewardUndo(Vec<Utxo>);
+
+impl BlockRewardUndo {
+    pub fn new(utxos: Vec<Utxo>) -> Self {
```

Comment:
I don't think this is sufficient. You need to also undo kernel spends in ConsensusData.

Reply:
Aren't kernel spends just inputs for BlockRewardTransactable?

Reply:
Yep. They're being spent to create a block, and the block rewards reflects their value + reward. So, they get marked "spent", and then we have to unspend them when disconnecting.

Reply:
Right, `BlockRewardUndo` stores this inputs and the `unspend_utxos_from_block_transactable` will add them to utxo set as unspent. I thought that I covered this, am I missing something?

Reply:
I'm not sure. Please make sure they're being unspent when disconnecting a block. I may have misunderstood how this is meant to be used. Please keep in mind that we don't have tests for this in chainstate for this yet.